src/os/bluestore/BlueStore.cc

```diff
@@ -6289,6 +6289,8 @@ void BlueStore::_txc_state_proc(TransContext *txc)
       txc->log_state_latency(logger, l_bluestore_state_prepare_lat);
       if (txc->ioc.has_pending_aios()) {
 	txc->state = TransContext::STATE_AIO_WAIT;
+	++txc->osr->txc_with_unstable_io;
+	txc->had_ios = true;
```

Comment:
If below ios can be done in submit_transaction_sync and KernelDevice.aio_thread are executing _txc_finish_io, holding osr->qlock

queue: (IO 1) (IO 2) 

ShardedOpWQ wants to deal with IO 3
If ShardedOpWQ context increments txc->osr->txc_with_unstable_io **before holding osr->qlock**, then IO 1 and IO 2 cannot be executed in submit_transaction_sync, because of txc->osr->txc_with_unstable_io incremented by later txn.

This is just my scenario.
If you have different opinions, I'm glad to receive them.

Thank you